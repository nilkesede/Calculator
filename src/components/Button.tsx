import React from 'react'
import styled from 'styled-components/native'

const Wrapper = styled.View<{
  readonly big?: boolean
  readonly dark?: boolean
  readonly primary?: boolean
}>`
  background-color: ${({theme}) => theme.colors.defaultGray}
  ${props => props.dark && `background-color: ${props.theme.colors.darkGray}`}
  ${props => props.primary && `background-color: ${props.theme.colors.primaryColor}`}

  flex: 1;
  ${({big}) => big && 'flex: 2;'}

  justify-content: center;
  margin: 1px;
`

const Text = styled.Text`
  font-weight: 200;
  font-size: 45px;
  color: white;
  text-align: center;
`

export default function Button({
  value,
  big = false,
  dark = false,
  primary = false
}: {
  readonly value: string
  readonly big?: boolean
  readonly dark?: boolean
  readonly primary?: boolean
}) {
  return (
    <Wrapper big={big} dark={dark} primary={primary}>
      <Text adjustsFontSizeToFit numberOfLines={1}>
        {value}
      </Text>
    </Wrapper>
  )
}
