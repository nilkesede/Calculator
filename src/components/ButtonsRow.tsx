import React from 'react'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  flex: 1;
  flex-direction: row;
`
export default function ButtonsRow({children}: {children: JSX.Element[]}) {
  return <Wrapper>{children}</Wrapper>
}
