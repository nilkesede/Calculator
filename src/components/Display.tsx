import React from 'react'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  background-color: #303030;
  flex: 1;
  justify-content: flex-end;
`

const Text = styled.Text`
  font-size: 100px;
  text-align: right;
  font-weight: 100;
  color: white;
`

export default function Display({value}: {value: string}) {
  return (
    <Wrapper>
      <Text adjustsFontSizeToFit numberOfLines={1}>
        {value}
      </Text>
    </Wrapper>
  )
}
