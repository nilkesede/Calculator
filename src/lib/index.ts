export {default as Processor} from './processor'
export {default as useProcessor} from './use-processor'
