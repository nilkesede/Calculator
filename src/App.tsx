import React from 'react'
import {StatusBar} from 'react-native'
import styled from 'styled-components/native'

import {Display, Keyboard} from './components'

const View = styled.View`
  flex: 1;
`

export default function App() {
  return (
    <>
      <StatusBar barStyle="light-content" />
      <View>
        <Display value={'123.456'} />
        <Keyboard />
      </View>
    </>
  )
}
