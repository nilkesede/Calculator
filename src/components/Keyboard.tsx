import React from 'react'
import styled from 'styled-components/native'

import ButtonsRow from './ButtonsRow'
import Button from './Button'

const Wrapper = styled.View`
  background-color: #303030;
  flex: 2;
`
export default function Keyboard() {
  return (
    <Wrapper>
      <ButtonsRow>
        <Button value={'AC'} />
        <Button value={'+/-'} />
        <Button value={'%'} />
        <Button value={'/'} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'7'} />
        <Button value={'8'} />
        <Button value={'9'} />
        <Button value={'x'} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'4'} />
        <Button value={'5'} />
        <Button value={'6'} />
        <Button value={'-'} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'1'} />
        <Button value={'2'} />
        <Button value={'3'} />
        <Button value={'+'} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'0'} big={true} />
        <Button value={'.'} />
        <Button value={'='} />
      </ButtonsRow>
    </Wrapper>
  )
}
