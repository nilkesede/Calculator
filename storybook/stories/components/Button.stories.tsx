import React from 'react'
import {storiesOf} from '@storybook/react-native'

import {Button} from '../../../src/components'

storiesOf('Button', module)
  .add('default', () => <Button value={'OK'} />)
  .add('big', () => <Button big value={'OK'} />)
  .add('dark', () => <Button dark value={'OK'} />)
  .add('primary', () => <Button primary value={'OK'} />)
