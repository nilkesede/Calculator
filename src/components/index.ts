import Button from './Button'
import ButtonsRow from './ButtonsRow'
import Display from './Display'
import Keyboard from './Keyboard'

export {Button, ButtonsRow, Display, Keyboard}
