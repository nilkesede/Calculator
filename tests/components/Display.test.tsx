import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import {Display} from '../../src/components'

it('renders correctly', () => {
  const tree = renderer.create(<Display value={'123.456'} />).toJSON()
  expect(tree).toMatchSnapshot()
})
