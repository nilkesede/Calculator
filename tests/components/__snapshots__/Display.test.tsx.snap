// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#303030",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 100,
          "fontWeight": "100",
          "textAlign": "right",
        },
      ]
    }
  >
    123.456
  </Text>
</View>
`;
